pkg/artifactory/resource/configuration/resource_artifactory_property_set.go

```diff
@@ -0,0 +1,290 @@
+package configuration
+
+import (
+	"context"
+	"github.com/jfrog/terraform-provider-shared/packer"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"gopkg.in/yaml.v3"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+type PredefinedValue struct {
+	Name         string `xml:"value" yaml:"-"`
+	DefaultValue bool   `xml:"defaultValue" yaml:"defaultValue"`
+}
+
+type Property struct {
+	Name                  string            `xml:"name" yaml:"-"`
+	PredefinedValues      []PredefinedValue `xml:"predefinedValues" yaml:"predefinedValues"`
+	ClosedPredefinedValue bool              `xml:"closedPredefinedValues" yaml:"closedPredefinedValues"`
+	MultipleChoice        bool              `xml:"multipleChoice" yaml:"multipleChoice"`
+}
+
+type PropertySet struct {
+	Name       string     `xml:"name" yaml:"-"`
+	Visible    bool       `xml:"visible" yaml:"visible"`
+	Properties []Property `xml:"properties>property" yaml:"properties"`
+}
+
+type PropertySets struct {
+	PropertySets []PropertySet `xml:"propertySets>propertySet" yaml:"propertySet"`
+}
+
+func ResourceArtifactoryPropertySet() *schema.Resource {
+	var propertySetsSchema = map[string]*schema.Schema{
+		"name": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      `Property set name.`,
+		},
+		"visible": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: `Defines if the list visible and assignable to the repository or artifact.`,
+		},
+		"property": {
+			Type:        schema.TypeSet,
+			Required:    true,
+			MinItems:    1,
+			Description: `A list of properties that will be part of the property set.`,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:             schema.TypeString,
+						Required:         true,
+						Description:      `The name of the property.`,
+						ValidateDiagFunc: validator.StringIsNotEmpty,
+					},
+					"closed_predefined_values": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+						Description: `Disables "multiple_choice" if set to "false" at the same time with multiple_choice set to "true".`,
+					},
+					"multiple_choice": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Default:     false,
+						Description: `Whether or not user can select multiple values. "closed_predefined_values" should be set to "true".`,
```

Comment:
We may want to use `CustomDiffFunc` to apply validation for this and `closed_predefined_values`.

Reply:
Added the function.